backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -129,14 +129,31 @@ def check_s3_iam_access(self) -> None:
                     f'{self.bucket_name}',
                 )
             )
+        else:
+            policy_check, missing_policies, extra_policies = share_policy_service.check_s3_actions(
+                existing_policy_statement=policy_document['Statement'][s3_statement_index]
+            )
+            if not policy_check:
+                logger.info(f'IAM Policy Statement {IAM_S3_BUCKETS_STATEMENT_SID}S3 has invalid actions')
+                self.bucket_errors.append(
+                    ShareErrorFormatter.invalid_policy_error_msg(
+                        self.target_requester_IAMRoleName,
+                        'IAM Policy Resource',
+                        f'{IAM_S3_BUCKETS_STATEMENT_SID}S3',
+                        'S3 Bucket',
+                        f'{self.bucket_name}',
+                        missing_actions=missing_policies,
+                        extra_actions=extra_policies,
+                    )
+                )
```

Comment:
need to `check_s3_actions(...)` for folder shares as well

Reply:
Such as with `backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_access_point_share_manager.py` file in https://github.com/data-dot-all/dataall/pull/1406/files#diff-1d48910db676c4a11b5ff22dcb45a55eb6c31ebd2d9aa1313f90ce4e13b8ecfc